lib/handleRequest.js

```diff
@@ -38,10 +38,16 @@ function handleRequest (req, res, params, context) {
   }
 
   if (method === 'OPTIONS' || method === 'DELETE') {
-    if (contentType === undefined) {
-      handler(reply)
-    } else {
+    if (
+      contentType !== undefined &&
+      (
+        headers['transfer-encoding'] !== undefined ||
+        headers['content-length'] !== undefined
+      )
+    ) {
       context.contentTypeParser.run(contentType, handler, request, reply)
+    } else {
+      handler(reply)
     }
```

Comment:
Why can't we merge this block with the one for POST?

Reply:
The logic is a little bit different between the types of requests.

The POST request logic will always parse the request body if a `Content-Type` is present, so if we used this same logic for DELETE requests, #936 wouldn't get fixed. The POST request logic will also try to parse the body when a `Content-Type` isn't present, which goes against the requirements for OPTIONS requests.